src/workerd/api/sockets.h

```diff
@@ -47,7 +47,8 @@ class Socket: public jsg::Object {
       jsg::Ref<ReadableStream> readableParam, jsg::Ref<WritableStream> writable,
       jsg::PromiseResolverPair<void> closedPrPair, kj::Promise<void> watchForDisconnectTask,
       jsg::Optional<SocketOptions> options, kj::Own<kj::TlsStarterCallback> tlsStarter,
-      bool isSecureSocket, kj::String domain, bool isDefaultFetchPort)
+      bool isSecureSocket, kj::String domain, bool isDefaultFetchPort,
+      jsg::PromiseResolverPair<void> openedPrPair)
```

Comment:
[nit] Perhaps `connectedPrPair` would be more consistent with existing terminology?

Reply:
`openedPrPair` is more consistent with `closed` and the upcoming attribute which we will add called `opened`.